src/NHibernate/Async/Transaction/AdoTransaction.cs

```diff
@@ -0,0 +1,92 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+
+using NHibernate.Engine;
+using NHibernate.Impl;
+
+namespace NHibernate.Transaction
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	/// <content>
+	/// Contains generated async methods
+	/// </content>
+	public partial class AdoTransaction : ITransaction
+	{
+
+		/// <summary>
+		/// Commits the <see cref="ITransaction"/> by flushing the <see cref="ISession"/>
+		/// and committing the <see cref="DbTransaction"/>.
+		/// </summary>
+		/// <param name="cancellationToken">A cancellation token that can be used to cancel the work</param>
+		/// <exception cref="TransactionException">
+		/// Thrown if there is any exception while trying to call <c>Commit()</c> on 
+		/// the underlying <see cref="DbTransaction"/>.
+		/// </exception>
+		public async Task CommitAsync(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
This case is a bit special. This public method may deserve some additional documentation, otherwise it could causes users to mis-believe NHibernate does commit asynchronously, and causes them to wonder why NHibernate does not have an async rollback.

Currently they are no way to commit or rollback asynchronously in .Net, be it with `DbTransaction` or system transactions. (Excepted with Npgsql, it has additional methods for that on its own `DbTransaction` implementation, the `NpgsqlTransaction`.)

Here this commit method is async in NHibernate because of the `Commit` flush mode, which may cause commit to call a flush, which can be async. But the commit itself remains synchronous.

Since the xml comment is taken from the synchronous method, I guess it will not be easy to add this information here. But it would be better if we could change the summary to something like:

```C#
/// Commits the <see cref="ITransaction"/> by flushing asynchronously the <see cref="ISession"/>
/// then committing synchronously the <see cref="DbTransaction"/>.
```

Conversely, we may have to add some warning about the `Commit` flush mode, when used with `TransactionScope`: in such case, the flush is always synchronous, since there is no way to asynchronously commit a transaction scope. (I have asked that [question](https://stackoverflow.com/q/44159992/1178314) about that subject, just to get in a comment argument with someone deeming it completely unneeded to have async for this. It got some upvotes probably because I have put a bounty on it, was a concern for some others, and I have widened its audience by adding a somewhat loosely related but high traffic tag, `c#`.)